pkg/ruby/check_upgrade.go

```diff
@@ -0,0 +1,199 @@
+package ruby
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v58/github"
+	"github.com/hashicorp/go-version"
+
+	"github.com/wolfi-dev/wolfictl/pkg/gh"
+)
+
+// CheckUpgrade attempts to find a gemspec for a given ruby gem and tries to
+// extract the required_ruby_version field from it. The required_ruby_version
+// should provide a set of constraints restricting the version of ruby a gem
+// can run on. CheckUpgrade simply tries to make sure the given RubyUpdateVersion
+// is within those constraints.
+func (o *Options) CheckUpgrade(pkg *Package) error {
+	// find the gemspec in the repository
+	gemspec, err := o.findGemspec(pkg)
+	if err != nil {
+		return fmt.Errorf("finding gemspec in repository: %w", err)
+	}
+
+	// download (and cache) the gemspec file
+	gemspecFile, err := o.fetchFile(pkg, gemspec)
+	if err != nil {
+		return fmt.Errorf("downloading gemspec: %w", err)
+	}
+
+	// search the gemspec for version constraints
+	return o.checkVersionConstraint(gemspecFile)
+}
+
+// findGemspec searches a given Github repository for a file named *.gemspec. It
+// searches for the file from the branch or tag specified in the melange yaml.
+func (o *Options) findGemspec(pkg *Package) (string, error) {
+	ctx := context.Background()
```

Comment:
I'm wondering if the context should be coming from the invoker.

Reply:
Yep, fixing :)